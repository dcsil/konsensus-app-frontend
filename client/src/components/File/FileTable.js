import React from 'react';

// reactstrap components
import {
  Card,
  CardHeader,
  CardFooter,
  Pagination,
  PaginationItem,
  PaginationLink,
  Table,
  Container,
  Row,
} from 'reactstrap';

import { useEffect, useState } from 'react';
import {
    getFiles,
  } from 'api/fileFunctions';


import FileRow from 'components/File/FileRow.js';

const FileTable = (props) => {

  const [starredFiles, setStarredFiles] = useState([]);

  useEffect(() => {
    getFiles('/file/starred', setStarredFiles);
  }, []);

  const handleStarred = (file) => {
    let isStarred = false;
    starredFiles.forEach((f) => {
      if (file.id === f.id) {
        isStarred = true;
      }
      });
      return isStarred;
    };

    return (
        <Container className="mt--7 pb-7" fluid>
        <Row>
          <div className="col">
            <Card className="shadow">
              <CardHeader className="border-0">
                <h3 className="mb-0">Files</h3>
              </CardHeader>

              <Table
                className="align-items-center table-flush"
                responsive
              >
                {/* </File> */}
                <thead className="thead-light">
                  <tr>
                    <th scope="col"></th>
                    <th scope="col">Name</th>
                    <th scope="col">Size</th>
                    <th scope="col">Collaborators</th>
                    <th scope="col">Last Updated</th>
                    <th scope="col" />
                  </tr>
                </thead>
                <tbody>
                {props.files.map((file) => {
                  const isStarred= handleStarred(file);
                  return ( 
                    <FileRow fileInfo={file}
                    key={file.id}
                    isStarred={isStarred}>
                    </FileRow>
                  );})}
              </tbody>
            </Table>

            <CardFooter className="py-4">
              <nav aria-label="...">
                <Pagination
                  className="pagination justify-content-end mb-0"
                  listClassName="justify-content-end mb-0"
                >
                  <PaginationItem className="disabled">
                    <PaginationLink
                      href="#pablo"
                      onClick={(e) => e.preventDefault()}
                      tabIndex="-1"
                    >
                      <i className="fas fa-angle-left" />
                      <span className="sr-only">Previous</span>
                    </PaginationLink>
                  </PaginationItem>
                  <PaginationItem className="active">
                    <PaginationLink
                      href="#pablo"
                      onClick={(e) => e.preventDefault()}
                    >
                      1
                    </PaginationLink>
                  </PaginationItem>
                  <PaginationItem className="disabled">
                    <PaginationLink
                      href="#pablo"
                      onClick={(e) => e.preventDefault()}
                    >
                      <i className="fas fa-angle-right" />
                      <span className="sr-only">Next</span>
                    </PaginationLink>
                  </PaginationItem>
                </Pagination>
              </nav>
            </CardFooter>
          </Card>
        </div>
      </Row>
    </Container>
  );
};

export default FileTable;
