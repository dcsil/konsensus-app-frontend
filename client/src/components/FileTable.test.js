import React from 'react';
import { shallow } from 'enzyme';
import FileTable from './File/FileTable';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('Test Sidebar component', () => {
  const wrapper = shallow(<FileTable />);
  expect(wrapper).toMatchSnapshot();
});
