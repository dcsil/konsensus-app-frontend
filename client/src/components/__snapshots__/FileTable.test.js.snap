// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FileTable component 1`] = `ShallowWrapper {}`;
